src/transform/src/reduction_pushdown.rs

```diff
@@ -24,7 +28,7 @@ impl crate::Transform for ReductionPushdown {
         relation: &mut MirRelationExpr,
         _: TransformArgs,
     ) -> Result<(), crate::TransformError> {
-        relation.visit_mut(&mut |e| {
+        relation.visit_mut_pre(&mut |e| {
```

Comment:
Having read through some later code, I think it makes sense to toss a comment in here (and we should probably do it generally in `transform` to explain the control flow, and why we use `_pre()` or not. In this case, I think it is that you want to chase after pushed down reductions.